experiments/prototype_omip_simulation/single_column_omip_simulation.jl

```diff
@@ -0,0 +1,406 @@
+
+    lines!(axu, times ./ days, interior(ua, 1, 1, 1, :))
+    lines!(axu, times ./ days, interior(va, 1, 1, 1, :))
+    lines!(axT, times ./ days, interior(Ta, 1, 1, 1, :))
+    lines!(axq, times ./ days, interior(qa, 1, 1, 1, :))
+
+    display(fig)
+    =#
+
+    sea_ice = nothing
+    radiation = Radiation()
+    coupled_model = OceanSeaIceModel(ocean, sea_ice; atmosphere, radiation)
+
+    coupled_simulation = Simulation(coupled_model, Δt=10minutes, stop_time=start_seconds + 60days)
+
+    elapsed = time_ns() - start_time
+    @info "Coupled simulation built. " * prettytime(elapsed * 1e-9)
+    start_time = time_ns()
+
+    wall_clock = Ref(time_ns())
+
+    function progress(sim)
+        msg = string("(", location, ")")
+        msg *= string(", iter: ", iteration(sim), ", time: ", prettytime(sim))
+
+        elapsed = 1e-9 * (time_ns() - wall_clock[])
+        msg *= string(", wall time: ", prettytime(elapsed))
+        wall_clock[] = time_ns()
+
+        u, v, w = sim.model.ocean.model.velocities
+        msg *= @sprintf(", max|u|: (%.2e, %.2e)", maximum(abs, u), maximum(abs, v))
+
+        T = sim.model.ocean.model.tracers.T
+        S = sim.model.ocean.model.tracers.S
+        e = sim.model.ocean.model.tracers.e
+
+        τˣ = first(sim.model.fluxes.total.ocean.momentum.τˣ)
+        τʸ = first(sim.model.fluxes.total.ocean.momentum.τʸ)
+        u★ = (τˣ^2 + τʸ^2)^(1/4)
+        Q = first(sim.model.fluxes.total.ocean.heat)
+
+        Nz = size(T, 3)
+        msg *= @sprintf(", u★: %.2f m s⁻¹", u★)
+        msg *= @sprintf(", Q: %.2f W m⁻²", Q)
+        msg *= @sprintf(", T₀: %.2f ᵒC",     first(interior(T, 1, 1, Nz)))
+        msg *= @sprintf(", extrema(T): (%.2f, %.2f) ᵒC", minimum(T), maximum(T))
+        msg *= @sprintf(", S₀: %.2f g/kg",   first(interior(S, 1, 1, Nz)))
+        msg *= @sprintf(", e₀: %.2e m² s⁻²", first(interior(e, 1, 1, Nz)))
+
+        @info msg
+    end
+
+    coupled_simulation.callbacks[:progress] = Callback(progress, IterationInterval(100))
+
+    # Build flux outputs
+    Jᵘ  = coupled_model.fluxes.total.ocean.momentum.u
+    Jᵛ  = coupled_model.fluxes.total.ocean.momentum.v
+    Jᵀ  = coupled_model.fluxes.total.ocean.tracers.T
+    Jˢ  = coupled_model.fluxes.total.ocean.tracers.S
+    E   = coupled_model.fluxes.turbulent.fields.water_vapor
+    Qse = coupled_model.fluxes.turbulent.fields.sensible_heat
+    Qla = coupled_model.fluxes.turbulent.fields.latent_heat
+    ρₒ  = coupled_model.fluxes.ocean_reference_density
+    cₚ  = coupled_model.fluxes.ocean_heat_capacity
+
+    Q = ρₒ * cₚ * Jᵀ
+    τˣ = ρₒ * Jᵘ
+    τʸ = ρₒ * Jᵛ
+    N² = buoyancy_frequency(ocean.model)
+    κᶜ = ocean.model.diffusivity_fields.κᶜ
+
+    fluxes = (; τˣ, τʸ, E, Jˢ, Q, Qse, Qla)
+
+    auxiliary_fields = (; N², κᶜ)
+    fields = merge(ocean.model.velocities, ocean.model.tracers, auxiliary_fields)
+
+    # Slice fields at the surface
+    outputs = merge(fields, fluxes)
+
+    output_attributes = Dict{String, Any}(
```

Comment:
let's open an issue or PR in Oceananigans to make the required changes so that the `{String, Any}` is not required here?

Reply:
do it. NetCDFOutputWriter probably needs a lot of work. I never use it.

Reply:
I just don't know what's needed to be modified. Is it these lines:

https://github.com/CliMA/Oceananigans.jl/blob/3b02321a350003fa9596c4f40839b031a5cc7ad1/src/OutputWriters/netcdf_output_writer.jl#L335-L337

should become

```Julia
                            global_attributes = Dict(String, Any),
                            output_attributes = Dict(String, Any),
                                   dimensions = Dict(String, Any),
```

Reply:
No. Those are the defaults in the constructor. The problem is wherever in the code that we assume `output_attributes` has some type.

Probably the problem is that we need to add elements to `output_attributes`. If the eltype of the dictionary is `Any`, we can add any element. If the eltype is something else, the `NetCDFOutputWriter` may error when trying to add it.

Thus the step that is needed is to convert the user-input into a type that will work with subsequent operations in `NetCDFOutputWriter`.